mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -1,61 +1,144 @@
+        xarray.DataSet.sel() for details on how this dictonary is used.
+        An example:
+            selectCorrdValues = {'cellLon': 180.0}
+
+    isel_values : dict, optional
+        A dictionary of coordinate names (keys) and indices, slices or
+        arrays of indices used to slice the variales in the data set.  See
+        xarray.DataSet.isel() for details on how this dictonary is used.
+        An example:
+            isel_values = {'nVertLevels': slice(0, 3),
+                           'nCells': cellIDs}
+
+    year_offset : float, optional
+        An offset used to convert an MPAS date to a date in the range supported
+        by xarray (numpy.datetime64).  Resulting dates must be between 1678
+        and 2622.
+
+    Returns
+    -------
+    ds : An xarray data set.
+
+    Raises
+    ------
+    TypeError
+        If the time variable has an unsupported type (not a date string,
+        a floating-pont number of days since the start of the simulation
+        or a numpy.datatime64 object).
+
+    ValueError
+        If the time variable is not found in the data set.
+
+    Author
+    ------
+    Xylar Asay-Davis
 
-def subset_variables(ds, vlist):  # {{{
+    Last modified
+    -------------
+    02/15/2017
     """
-    Reduces an xarray dataset ds to only contain the variables in vlist.
 
-    Phillip J. Wolfram
-    01/10/2017
+    def preprocess_fn(ds):
+        return preprocess(ds,
+                          time_variable_name=time_variable_name,
+                          variable_list=variable_list,
+                          sel_values=sel_values,
+                          isel_values=isel_values,
+                          year_offset=year_offset)
+
+    ds = xarray.open_mfdataset(file_names,
+                               preprocess=preprocess_fn,
+                               decode_times=False, concat_dim='Time')
+
+    ds = remove_repeated_time_index(ds)
+
+    return ds  # }}}
+
+
+def subset_variables(ds, variable_list):  # {{{
+    """
+    Given a data set and a list of variable names, returns a new data set that
+    contains only variables with those names.
+
+    Parameters
+    ----------
+    ds : xarray.DataSet object
+        The data set from which a subset of variables is to be extracted.
+
+    variable_list : string or list of strings
+        The names of the variables to be extracted.
+
+    Returns
+    -------
+    ds : xarray.DataSet object
+        A copy of the original data set with only the variables in
+        variable_list.
+
+    Raises
+    ------
+    ValueError
+        If the resulting data set is empty.
+
+    Authors
+    -------
+    Phillip J. Wolfram, Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/10/2017
     """
 
     allvars = ds.data_vars.keys()
 
     # get set of variables to drop (all ds variables not in vlist)
-    dropvars = set(allvars) - set(vlist)
+    dropvars = set(allvars) - set(variable_list)
```

Comment:
Can we have some type of local document on standards in this repo?  It will be easier to refer to if it is local to this repository and not on confluence, although a mirror on confluence may be optimal.

@xylar, you are right that underscored variable names are used in xarray.  I would personally say that is suboptimal for us, particularly since we are allowing use of camel case for variable names.  I'm happy to bend to the group if you and @milenaveneziani find this too restrictive, however.